src/unix/poll.c

```diff
@@ -32,9 +32,8 @@ static void uv__poll_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
   int pevents;
 
   handle = container_of(w, uv_poll_t, io_watcher);
-
-  if (events & POLLERR) {
-    uv__io_stop(loop, w, POLLIN | POLLOUT | UV__POLLRDHUP);
+  if ((events & POLLERR) && (events & UV__POLLPRI) != UV__POLLPRI) {
```

Comment:
Some more documentation on the 'issue':
http://stackoverflow.com/questions/27411013/poll-returns-both-pollpri-pollerr

It happens inside the kernel `fs/kernfs/file.c`:
> 775 /*
776  * Kernfs attribute files are pollable.  The idea is that you read
777  * the content and then you use 'poll' or 'select' to wait for
778  * the content to change.  When the content changes (assuming the
779  * manager for the kobject supports notification), **poll will
780  * return POLLERR|POLLPRI**, and select will return the fd whether
781  * it is waiting for read, write, or exceptions.
782  * Once poll/select indicates that the value has changed, you
783  * need to close and re-open the file, or seek to 0 and read again.
784  * Reminder: this only works for attributes which actively support
785  * it, and it is not possible to test an attribute from userspace
786  * to see if it supports poll (Neither 'poll' nor 'select' return
787  * an appropriate error code).  When in doubt, set a suitable timeout value.
788  */

The haskell source comments are also helpfull:
https://hackage.haskell.org/package/hpio-0.8.0.3/src/src/System/GPIO/Linux/Sysfs/pollSysfs.c

TLDR: sysfs POLLPRI polling is not going to work without the additional condition we are discussing here, as per linux kernel implementation.